src/weekdays.rs

```diff
@@ -0,0 +1,723 @@
+    fn bitand_assign(&mut self, rhs: Self) {
+        self.0 &= rhs.0;
+    }
+}
+impl BitXorAssign for Weekdays {
+    fn bitxor_assign(&mut self, rhs: Self) {
+        self.0 ^= rhs.0;
+    }
+}
+
+impl Not for Weekdays {
+    type Output = Self;
+
+    fn not(self) -> Self::Output {
+        self.inverse()
+    }
+}
+#[cfg(test)]
+#[test]
+fn not_operation_preserves_8th_bit_invariant() {
+    for weekdays in Weekdays::iter_all() {
+        (!weekdays).assert_8th_bit_invariant();
+    }
+}
+
+impl From<Weekday> for Weekdays {
+    fn from(weekday: Weekday) -> Self {
+        Self::from_one(weekday)
+    }
+}
+
+impl Extend<Weekday> for Weekdays {
+    fn extend<T: IntoIterator<Item = Weekday>>(&mut self, iter: T) {
+        for weekday in iter {
+            self.insert(weekday);
+        }
+    }
+}
+impl FromIterator<Weekday> for Weekdays {
+    fn from_iter<T: IntoIterator<Item = Weekday>>(iter: T) -> Self {
+        let mut weekdays = Self::EMPTY;
+        weekdays.extend(iter);
+        weekdays
+    }
+}
+
+impl IntoIterator for Weekdays {
+    type Item = Weekday;
+    type IntoIter = WeekdaysIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
+
+// impl Bit*<Weekday> for Weekdays
+impl BitOr<Weekday> for Weekdays {
+    type Output = Self;
+
+    fn bitor(self, rhs: Weekday) -> Self::Output {
+        self | Self::from(rhs)
+    }
+}
+impl BitAnd<Weekday> for Weekdays {
+    type Output = Self;
+
+    fn bitand(self, rhs: Weekday) -> Self::Output {
+        self & Self::from(rhs)
+    }
+}
+impl BitXor<Weekday> for Weekdays {
+    type Output = Self;
+
+    fn bitxor(self, rhs: Weekday) -> Self::Output {
+        self ^ Self::from(rhs)
+    }
+}
+
+/// Can be used to check the presence of a day in the collection.
+///
+/// # Example
+/// ```
+/// # use chrono::{Weekday, Weekdays};
+/// assert!(Weekdays::MON[Weekday::Mon]);
+/// assert!(Weekdays::ALL[Weekday::Mon]);
+/// assert!(!Weekdays::EMPTY[Weekday::Mon]);
+/// assert!(!Weekdays::TUE[Weekday::Mon]);
+/// ```
+impl Index<Weekday> for Weekdays {
+    type Output = bool;
+
+    fn index(&self, weekday: Weekday) -> &Self::Output {
+        if self.contains(weekday) { &true } else { &false }
+    }
+}
+
+// impl Bit*<Weekdays> for Weekday
+impl BitOr<Weekdays> for Weekday {
+    type Output = Weekdays;
+
+    fn bitor(self, rhs: Weekdays) -> Self::Output {
+        Weekdays::from(self) | rhs
+    }
+}
+impl BitAnd<Weekdays> for Weekday {
+    type Output = Weekdays;
+
+    fn bitand(self, rhs: Weekdays) -> Self::Output {
+        Weekdays::from(self) & rhs
+    }
+}
+impl BitXor<Weekdays> for Weekday {
+    type Output = Weekdays;
+
+    fn bitxor(self, rhs: Weekdays) -> Self::Output {
+        Weekdays::from(self) ^ rhs
+    }
+}
+
+// impl Bit*Assign<Weekday> for Weekdays
+impl BitOrAssign<Weekday> for Weekdays {
+    fn bitor_assign(&mut self, rhs: Weekday) {
+        *self |= Self::from(rhs);
+    }
+}
+impl BitAndAssign<Weekday> for Weekdays {
```

Comment:
Some of these look a little weird in isolation.

`days &= day` means "remove all weekdays other than `day` from `days`". I would be hard pressed to come up with a use case for this one.

On the other hand, the meaning is still clear. And one would probably be surprised to _not_ be able to do this when other similar operations are available.

Reply:
Let's drop all the `BitX` impls for now and come back to them later.